### PR TITLE
[camera] Support pausing and resuming preview

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Addresses #29208

# How
Added functions to pause and resume the preview

# Test Plan
Updated the example screen, adding a play/pause button. It's working as expected

### CODE REVIEW

File: packages/expo-notifications/android/src/test/java/expo/modules/notifications/service/delegates/ExpoHandlingDelegateTest.kt [hunk 1]
```diff
@@ -101,7 +101,7 @@ class ExpoHandlingDelegateTest {
         NotificationContent.Builder().setTitle(title).setText(text).setBody(body).build(),
         object : NotificationTrigger {
           override fun describeContents(): Int = -1
-          override fun writeToParcel(dest: Parcel?, flags: Int) {}
+          override fun writeToParcel(dest: Parcel, flags: Int) {}
```
Comment (alanjhughes) [inline packages/expo-notifications/android/src/test/java/expo/modules/notifications/service/delegates/ExpoHandlingDelegateTest.kt:104]: Ran into a compiler error here. Parcel is not defined as nullable in the interface

File: packages/expo-camera/src/CameraView.tsx [hunk 2]
```diff
@@ -116,6 +116,21 @@ export default class CameraView extends Component<CameraProps> {
     return (await this._cameraRef.current?.getAvailablePictureSizes()) ?? [];
   }
 
+  /**
+   * Resumes the camera preview.
+   */
+
```
Comment (tsapeta) [inline packages/expo-camera/src/CameraView.tsx:122]:
```suggestion
```

File: packages/expo-camera/src/CameraView.tsx [hunk 3]
```diff
@@ -116,6 +116,21 @@ export default class CameraView extends Component<CameraProps> {
...
+  async resumePreview(): Promise<void> {
+    return this._cameraRef.current?.resumePreview();
+  }
+
+  /**
+   * Pauses the camera preview. It is not recommended to use `takePictureAsync` when preview is paused.
```
Comment (tsapeta) [inline packages/expo-camera/src/CameraView.tsx:128]: What happens when you take the picture when it's paused? Maybe we should note this in `takePictureAsync` as well.

**Review by tsapeta — APPROVED** [review]

File: packages/expo-camera/src/CameraView.tsx (hunk 3 above)
Comment (alanjhughes) [reply packages/expo-camera/src/CameraView.tsx:127]: Done 👍

**Review by douglowder — APPROVED** [review]

**Review by lukmccall — APPROVED** [review]

**Pull request merged**